### ISSUE TITLE
Docstring code examples should execute

### ISSUE THREAD

**Issue by crantila** [ELVIS-Project/vis-framework#319]
The docstring code examples aren't executable code. They should be.

**Comment by musicus** [comment]
Always!

**Comment by mborsodi** [comment]
All the docstring examples? Or are there specific ones?

**Comment by crantila** [comment]
All. They needn't be particularly useful, but the expectation of experienced Python users is that any sample code in documentation should be executable.

**Comment by musicus** [comment]
Yes, all.

**Comment by mborsodi** [comment]
Also, what parts of the code should actually have examples? Right now it looks like 1) mostly only run() has examples, which makes sense because it is the part people will be interacting with the most when using VIS and 2) there are a couple of indexer_func() with examples, but it's pretty unlikely that anybody not working on VIS is actually going to need that. So, am I making these examples for users or VIS developers?

**Comment by musicus** [comment]
Proceed at what we just discussed.

**Comment by crantila** [comment]
What did you just discuss?

My model for good API documentation, for a long time, has been Qt. See [this](https://doc.qt.io/qt-5/qml-qtwebengine-webenginehistory.html) as an example. The idea is this:
- one-sentence description
- parameter type and description
- if the function raises any exceptions, mention this
- further prose description, if required
- an example, or some examples, depending on the complexity of the function

**Comment by mborsodi** [comment]
OK, I added examples in everything except for:

- ngram (because it's deprecated)
- dissonance (the tests for this are failing for me, so I'll add it as soon as I figure this out)
- lilypond, dendrogram, barchart (are these going to be used very much anymore once the VIS-Ualizer is ready?)

**Comment by musicus** [comment]
I'm assuming you did the new_ngram? Ok on dissonance indexer, and Point 3, yes VIS-Ualizer will overtake these. 😸

**Comment by mborsodi** [comment]
Yes, new_ngram is done. So I can close the issue once I do dissonance indexer? Or is there anything else I'm forgetting?

**Comment by mborsodi** [comment]
Done, in my branch.

**Comment by musicus** [comment]
It should only close when it is merged into main ... if you set it up appropriately the issue will be closed automatically at that point. Look it up.
